### PR TITLE
Fix issue 98: rerun method uses same class

### PR DESCRIPTION
**Pull request by aledsage**
- if run test class MySub, and then select + rerun MySub.mymethod
  inherited from MyAbstractSuper, then it will run MySub.mymethod
  again, rather than trying to run MyAbstractSuper.mymethod

### CODE REVIEW

**Comment by missedone** [comment]
@aledsafe let's revisit this PR, do you still have issue #98 with latest vesion of testng eclipse plugin. if so, could you update this PR so that we can have a look then
thanks

**Comment by aledsage** [comment]
@missedone Thanks for looking at this! I confirmed #98 was still an issue on master, and have re-applied my changes again master. PR is again ready for review.

File: testng-eclipse-plugin/src/main/org/testng/eclipse/util/LaunchUtil.java [hunk 1]
```diff
@@ -284,19 +284,30 @@ public static void launchMethodConfiguration(IJavaProject javaProject,
...
+  }
+  
+  public static void launchMethodConfiguration(IJavaProject javaProject,
+          IType iType,
+          IMethod iMethod,
+          String runMode,
+          RunInfo runInfo) {
 
-    Set<IMethod> methods = Sets.newHashSet(iMethod);
+    Set<TypeAndMethod> typeAndMethods = Sets.newHashSet(new TypeAndMethod(iType, iMethod));
```
Comment (missedone) [inline testng-eclipse-plugin/src/main/org/testng/eclipse/util/LaunchUtil.java]: now that `TypeAndMethod` is stored in `Set`, you need to override `equals()` and `hashCode()`

**Comment by missedone** [comment]
hi @aledsage 
looks good except one minor issue in my comment.

Thanks &
Happy New Year! :tada:

**Comment by aledsage** [comment]
@missedone oops, thanks for spotting missing hashCode/equals! I've added those.

Note I'd have preferred to use the new Java 7 utilities:

    return Objects.hash(type, method);
    return Objects.equals(type, o.type) && Objects.equals(method, o.method);

But nothing else in the code uses that, so I went with consistency. Does testng-eclipse require Java 7 or later?

**Comment by missedone** [comment]
hi @aledsage , 
please go ahead to use Java 7 APIs, Java 7 is the minimum required version.
thanks

**Comment by aledsage** [comment]
Thanks @missedone - changed to use `Objects.hash` etc (rebased to change the second commit accordingly).

**Comment by missedone** [comment]
thanks @aledsage

**Pull request merged**